### PR TITLE
feat(ui): adding truncate and line clamp option

### PR DESCRIPTION
**Pull request by BillyFigueroa**
## Description
Adding the ability to set widths on columns (i.e. table cells) and truncate the text using ellipsis and overflow.

Also adding a limit of lines a table cell can display of 4 using line clamp for the `Applications > Packages` view

## Related Issue

- #273 

#### Before
Screenshot 2024-09-04 at 12 04 45 PM

#### After
Screenshot 2024-09-04 at 11 59 55 AM

#### Before
Screenshot 2024-09-04 at 12 02 54 PM

#### After
Screenshot 2024-09-04 at 12 02 25 PM

### CODE REVIEW

**Comment by UncleGedd** [comment]
Nice work! For the Packages view, I noticed on the first row that only 1 component is displayed. If the line clamp is set to 3, shouldn't we be seeing 3 components before the clamp?

**Comment by BillyFigueroa** [comment]
> Nice work! For the Packages view, I noticed on the first row that only 1 component is displayed. If the line clamp is set to 3, shouldn't we be seeing 3 components before the clamp?

Were you seeing that locally or are you referring to the screenshot in the description for the PR?

**Comment by UncleGedd** [comment]
> > Nice work! For the Packages view, I noticed on the first row that only 1 component is displayed. If the line clamp is set to 3, shouldn't we be seeing 3 components before the clamp?
> 
> Were you seeing that locally or are you referring to the screenshot in the description for the PR?

Just looking at the screenshot, I haven't run it locally yet

File: ui/tests/navigation.spec.ts [hunk 1]
```diff
@@ -5,6 +5,11 @@ import { expect, test } from '@playwright/test'
 
 test.describe('Navigation', async () => {
   test.beforeEach(async ({ page }) => {
+    await page.setViewportSize({
```
Comment (TristanHoladay) [inline ui/tests/navigation.spec.ts:8]: just curious if there's a reason for this specific sizing

Comment (BillyFigueroa) [reply ui/tests/navigation.spec.ts:8]: > just curious if there's a reason for this specific sizing

It is the 14" MacBook Pro. No real reason. I just googled it's size and used that. Could have been smaller or bigger

File: ui/src/lib/components/Tooltip/component.svelte [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+      // add the newly created element and its content into the DOM
+      const bodyEl = document.querySelector('body')!
+      bodyEl.appendChild(newEl)
+      const createdEL = document.getElementById('test-el')
+      const createElWidth = createdEL?.clientWidth
+      bodyEl.removeChild(newEl)
+
+      return createElWidth || 0
+    }
```
[the remark this answers is not in the document]
Comment (BillyFigueroa) [reply ui/src/lib/components/Tooltip/component.svelte:59]: I ll check it out

[the remark this answers is not in the document]
Comment (BillyFigueroa) [reply ui/src/lib/components/Tooltip/component.svelte:59]: @TristanHoladay I just tried and this is focused on the content itself i.e. the text. What I am figuring out here is the size of a div that would fit the text inside and compared that to the current width of the table cell. Based off that I determine if the cell is bigger or smaller and if it needs to be clipped

[the remark this answers is not in the document]
Comment (TristanHoladay) [reply ui/src/lib/components/Tooltip/component.svelte:59]: ah, gotcha. thanks for checking if that'd work.

**Review by TristanHoladay — APPROVED** [review]

**Pull request merged**